lib/simple-auth-policy.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const crypto = require('crypto');
+const EventEmitter = require('events').EventEmitter;
+
+const errors = require('./errors');
+
+// Simple generic authentication provider. You are free to implement
+// whatever suits your needs instead.
+//
+module.exports = class SimpleAuthPolicy extends EventEmitter {
+  constructor() {
+    super();
+  }
+
+  // Start session. Only anonymous handshakes are allowed.
+  //   connection - JSTP connection
+  //   application - application instance
+  //   strategy - authentication strategy (only 'anonymous' is supported)
+  //   credentials - authentication credentials
+  //   callback - callback function that has signature
+  //              (error, username, sessionId)
+  //
+  startSession(connection, application, strategy, credentials, callback) {
+    if (strategy !== 'anonymous') {
+      return callback(errors.ERR_AUTH_FAILED);
+    }
+
+    const sessionId = generateUuid4();
+    this.emit('session', sessionId, connection, application);
+
+    callback(null, null, sessionId);
+  }
+};
+
+// Generate UUID v4
+//
+function generateUuid4() {
```

Comment:
This function shouldn't be in this package, it falls outside its scope. While you are here and doing some refactoring, maybe we should drop in this PR? Exactly this function (which I replaced the one from Impress with at some point) is now in `metarhia-common`, I believe; and there are other npm packages too, like `uuid`.